.github/workflows/compilation.yml

```diff
@@ -0,0 +1,50 @@
+name: CI-compile
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    # container: ghcr.io/${{ github.repository_owner }}/mipsr5900el-gentoo-linux-gnu:main
+    container: ghcr.io/AKuHAK/mipsr5900el-gentoo-linux-gnu:main
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Compile project
+        env:
+          INSTALL_MOD_PATH: ../initramfs/ps2/
+          INSTALL_MOD_STRIP: 1
+        run: |
+          mv -rf ../svr/iniramfs ../
+          make -j $(getconf _NPROCESSORS_ONLN) ps2_defconfig
+          make -j $(getconf _NPROCESSORS_ONLN) oldconfig
+          make -j $(getconf _NPROCESSORS_ONLN) vmlinux
+          make -j $(getconf _NPROCESSORS_ONLN) modules
+          make -j $(getconf _NPROCESSORS_ONLN) modules_install
+          make -j $(getconf _NPROCESSORS_ONLN) vmlinuz
+          cp vmlinuz PS2Linux5.4-${{ github.sha }}.ELF
+
+      - name: Upload initramfs as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: initramfs
+          path: initramfs
+
+      - name: Upload kernel as artifact
+        if: ${{ success() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: "PS2Linux5.4-${{ github.sha }}.ELF"
```

Comment:
Is it possible to replace `5.4` with the branch name, and shorten `${{ github.sha }}` to the first 8 characters or so? The branch seems to be something like `github.ref`, according to https://docs.github.com/en/enterprise-server@3.2/actions/learn-github-actions/contexts.

Git Hub doesn't seem to have a substring function, but there are apparently workarounds like https://github.community/t/substring-function/18401.

So assuming the branch is `ps2-main` it'd be something like `vmlinuz-ps2-main-dc19f33e.elf`.

Reply:
> Is it possible to replace 5.4 with the branch name

Yes of course.

As about sha. I used some tricks for making sha shorter, I just dont like that for such simple action we need a workarounds that require 3-4 complex code texting. But if you think that we need this, I can revisit this part.

Reply:
Right. Let's revisit shortened sha:s when circumstances have improved.